responders/src/UI/embc-responder/src/app/feature-components/search/essfile-dashboard/ess-file-details/ess-file-details.component.spec.ts

```diff
@@ -1,27 +1,224 @@
+    securityPhraseEdited: false,
+    isRestricted: false,
+    status: EvacuationFileStatus.Active,
+    evacuationFileDate: '2021-11-23T23:32:16Z',
+    householdMembers: [
+      {
+        id: '42ab5087-ce1c-48ec-b339-1b1c173842d7',
+        linkedRegistrantId: 'a677e7e1-54ea-48ef-a030-b48c56fccd9c',
+        firstName: 'Anne',
+        lastName: 'Lee',
+        initials: null,
+        gender: 'Female',
+        dateOfBirth: '09/09/1999',
+        type: HouseholdMemberType.Registrant,
+        isPrimaryRegistrant: true,
+        isHouseholdMember: false,
+        isUnder19: false,
+        isRestricted: false,
+        isVerified: true
+      },
+      {
+        id: 'c0350e7e-055b-45c9-b32d-1c81c14afdd5',
+        linkedRegistrantId: null,
+        firstName: 'abcd',
+        lastName:
+          'erfehrbfjherbfjhrebfjherhbfnceujkrsdhcnqeukasjdcnesjka,dnukasj,dmwehnfurgfyrgfgerhfjdvfdhjvbfdjhbvhd',
+        initials: null,
+        gender: 'Female',
+        dateOfBirth: '09/09/1999',
+        type: HouseholdMemberType.HouseholdMember,
+        isPrimaryRegistrant: false,
+        isHouseholdMember: true,
+        isUnder19: false,
+        isRestricted: null,
+        isVerified: null
+      }
+    ],
+    task: {
+      taskNumber: 'UNIT-TEST-ACTIVE-TASK',
+      communityCode: '9e6adfaf-9f97-ea11-b813-005056830319',
+      from: '2021-05-12T21:31:00Z',
+      to: '2022-09-08T18:53:00Z'
+    },
+    assignedTaskCommunity: {
+      code: '9e6adfaf-9f97-ea11-b813-005056830319',
+      name: 'Victoria',
+      districtName: 'Capital',
+      stateProvinceCode: 'BC',
+      countryCode: 'CAN',
+      type: CommunityType.City
+    }
+  };
+  const routerMock = {
+    navigate: jasmine.createSpy('navigate'),
+    events: of(new NavigationStart(1, 'regular')),
+    getCurrentNavigation: () => ({
+      extras: { state: mockEssfileState }
+    })
+  };
 
   beforeEach(async () => {
     await TestBed.configureTestingModule({
-      imports: [RouterTestingModule, CustomPipeModule],
-      declarations: [EssFileDetailsComponent]
+      imports: [RouterTestingModule, CustomPipeModule, HttpClientTestingModule],
+      declarations: [EssFileDetailsComponent],
+      providers: [
+        EssFileDetailsComponent,
+        {
+          provide: EssfileDashboardService,
+          useClass: MockEssfileDashboardService
+        },
+        {
+          provide: Router,
+          useValue: routerMock
+        }
+      ]
     }).compileComponents();
   });
 
   beforeEach(() => {
     fixture = TestBed.createComponent(EssFileDetailsComponent);
     component = fixture.componentInstance;
+    essfileDashboardService = TestBed.inject(EssfileDashboardService);
+    essfileDashboardService.essFile = mockEssfile;
+  });
+
+  it('should create', () => {
     fixture.detectChanges();
+    expect(component).toBeTruthy();
   });
 
-  // it('should create', () => {
-  //   expect(component).toBeTruthy();
-  // });
+  it('should get essFile from router', () => {
```

Comment:
What's the difference between this and the next test case and what is mockEssfileState and how is this being used?